File: frontend/src/api/dto/rideListResponse.ts
import type { RideDto } from './rideDto.ts'

/**
 * Paginated ride list response
 */
export interface RideListResponse {
  /** List of rides */
  rides: RideDto[]
  /** Total number of rides */
  total: number
  /** Current page number */
  page: number
  /** Page size */
  size: number
}

File: backend/src/main/java/fr/pedalons/dto/rides/response/RideListResponse.java
package fr.pedalons.dto.rides.response;

import fr.pedalons.dto.validation.ValidateSchema;
import java.util.List;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

@Schema(description = "Paginated ride list response")
@ValidateSchema
public record RideListResponse(
    @Schema(description = "List of rides", required = true) List<RideDto> rides,
    @Schema(description = "Total number of rides", required = true) long total,
    @Schema(description = "Current page number", required = true) int page,
    @Schema(description = "Page size", required = true) int size) {}
